consensus/src/round_manager.rs

```diff
@@ -531,8 +560,34 @@ impl RoundManager {
         );
 
         observe_block(proposal.timestamp_usecs(), BlockStage::SYNCED);
+        self.process_verified_proposal(proposal).await
+    }
 
+    async fn resend_verified_proposal_to_self(&self, proposal: Block, interval_ms: u64) {
+        let author = self.network.author();
+        let self_sender = self.round_manager_tx.clone();
+        let event = VerifiedEvent::VerifiedProposalMsg(Box::new(proposal));
+        tokio::spawn(async move {
+            sleep(Duration::from_millis(interval_ms)).await;
```

Comment:
I'd prefer to take in Arc<BlockStore> and check the back pressure in a loop until it's resolved before sending it back

Reply:
Changed.